src/System.Management.Automation/engine/SessionStateScopeAPIs.cs

```diff
@@ -175,15 +175,17 @@ internal SessionStateScope CurrentScope
 
             set
             {
-                Diagnostics.Assert(
-                    value != null,
-                    "A null scope should never be set");
+                if (value == null)
+                {
+                     throw PSTraceSource.NewArgumentException(nameof(CurrentScope));
+                }
+
```

Comment:
Do we have an official guideline on where it is more appropriate to use `Diagnostics.Assert()` and where it's more appropriate to `throw`? I'd be hesitant to change them over without good reason.

Reply:
We run all tests  (CIs) only on Release configuration. So Diagnostics.Assert() is not so useful.